### DISCUSSION TITLE
Override default ResultHandler

### DISCUSSION THREAD

**Discussion by pd-gov** [spring-projects/spring-shell#683] [General]
It seems that you can't override the default `ResultHandler` while using the auto-configuration.

I would like to register a `ResultHandler` as the default or at least a way to convert result values into a String-representation. My goal is to get json output.

**Comment by jvalkeal** [comment]
Is this for cases where you'd like to process output programmatically?

While we don't currently allow overriding you can add your own `ResultHandler` beans and use ordering from framework like `@Order`. Maybe that would suit your needs.

**Reply by chisui** [reply]
1. Yes, this is for processing output programmatically. Ideally I would like a `--format/-t (json|plain)` parameter to allow for both human readable and processable output.
2. I tried creating a `ResultHandler<Object>` bean with an `@Order(Ordered.HIGHEST_PRECEDENCE)` and it wasn't used.

**Reply by jvalkeal** [reply]
Ah yeah it's probably because some other type is resolved first and `Object` type is last as a generic type. It would be super nice to support things like `--json` or `--yaml`, etc, even so that you'd have those a global option(when we actually get support for global options). Lemmy play with these things to see what's possible now and what kind of enhancements we'd need to do in a shell itself.
